### PR TITLE
Update recommended list of MET filters

### PR DESCRIPTION
**Pull request by lathomas**
#### PR description:
The PR update the list of MET filters run in the global boolean Flag_METFilters.
The list was discussed here:
https://indico.cern.ch/event/1042496/#27-update-on-ul-met-filters
and corresponds to the list in this twiki:
https://twiki.cern.ch/twiki/bin/viewauth/CMS/MissingETOptionalFiltersRun2#Analysis_Recommendations_for_ana

#### PR validation:
runTheMatrix.py -l limited -i all --ibeos
is run successfully

### CODE REVIEW

**Comment by qliphy** [comment]
please test

**Comment by slava77** [comment]
it looks like the failures were related to execution with this PR code.
eg. in 136.7722

```
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 283877 lumi: 11 event: 17578209 stream: 0
   [1] Running path 'nanoAOD_step'
   [2] Calling method for module GlobalVariablesTableProducer/'extraFlagsTable'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: bool
Looking for module label: BadChargedCandidateTagger
Looking for productInstanceName: 

```
please fix.

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoMET/METFilters/python/metFilters_cff.py [hunk 1]
```diff
@@ -95,39 +95,23 @@
...
-#   trkPOGFilters
-   chargedHadronTrackResolutionFilter *
-   BadChargedCandidateFilter*
-   BadPFMuonFilter *
-   BadPFMuonDzFilter *
-   hfNoisyHitsFilter *
-   BadChargedCandidateSummer16Filter*
-   BadPFMuonSummer16Filter *
-   muonBadTrackFilter
+    goodVertices *
```
Comment (perrotta) [inline RecoMET/METFilters/python/metFilters_cff.py:98]: Who needs `goodVertices` now? (It was supposed to be needed by the tracking failure filter, which is not in the list)

File: RecoMET/METFilters/python/metFilters_cff.py [hunk 2]
```diff
@@ -95,39 +95,23 @@
...
+    globalSuperTightHalo2016Filter *
```
Comment (perrotta) [inline RecoMET/METFilters/python/metFilters_cff.py:99]: This replaces `CSCTightHaloFilter`: correct?

**Review by perrotta — COMMENTED** [review]
I am missing where this PR "disables the recomputation on MINIAOD of some filter decisions that already exist": did you mean "nanoAOD" instead, and this part is what was reverted with your last commit?
In any case, please make sure that the PR description corresponds to what implemented in the PR, and modify it if needed

File: RecoMET/METFilters/python/metFilters_cff.py (hunk 1 above)
Comment (lathomas) [reply RecoMET/METFilters/python/metFilters_cff.py:98]: The filter is still needed. It basically checks whether there's at least one good vertex is the event which help rejecting some noise.

File: RecoMET/METFilters/python/metFilters_cff.py (hunk 2 above)
Comment (lathomas) [reply RecoMET/METFilters/python/metFilters_cff.py:99]: Yes

**Comment by perrotta** [comment]
@lathomas :
- I see you updated the PR description (thank you), could you please also update the title if "disable recomputation on MINIAOD" does not apply any more?
- Could you also please post some presentation (at the JetMET POG, or somewhere else) where this updated list of MET filters was presented and approved?

**Comment by lathomas** [comment]
Sorry I thought I updated the title too. Done now. 
As for the documentation, the list was discussed here: 
https://indico.cern.ch/event/1042496/#27-update-on-ul-met-filters
and corresponds to the list in this twiki: 
https://twiki.cern.ch/twiki/bin/viewauth/CMS/MissingETOptionalFiltersRun2#Analysis_Recommendations_for_ana

File: RecoMET/METFilters/python/metFilters_cff.py [hunk 3]
```diff
@@ -95,39 +95,23 @@
...
+    HBHENoiseFilterResultProducer  *
+    HBHENoiseFilter *
+    HBHENoiseIsoFilter *
+    EcalDeadCellTriggerPrimitiveFilter *
+    BadPFMuonFilter *
+    BadPFMuonDzFilter *
+    EcalDeadCellTriggerPrimitiveFilter *
```
Comment (perrotta) [inline RecoMET/METFilters/python/metFilters_cff.py:106]: This is repeated twice

File: RecoMET/METFilters/python/metFilters_cff.py [hunk 4]
```diff
@@ -95,39 +95,23 @@
...
+    goodVertices *
+    globalSuperTightHalo2016Filter *
+    HBHENoiseFilterResultProducer  *
+    HBHENoiseFilter *
+    HBHENoiseIsoFilter *
+    EcalDeadCellTriggerPrimitiveFilter *
+    BadPFMuonFilter *
+    BadPFMuonDzFilter *
+    EcalDeadCellTriggerPrimitiveFilter * 
+    ecalBadCalibFilter *
```
Comment (perrotta) [inline RecoMET/METFilters/python/metFilters_cff.py:107]: According to the [twiki](https://twiki.cern.ch/twiki/bin/viewauth/CMS/MissingETOptionalFiltersRun2#Analysis_Recommendations_for_ana) you point out, this is not among the recommended filters for 2016 data and MC (UL): do you plan to remove it with a modifier?

File: RecoMET/METFilters/python/metFilters_cff.py [hunk 5]
```diff
@@ -95,39 +95,23 @@
...
-   ecalBadCalibFilter*
-#   *goodVertices * trackingFailureFilter *
-   eeBadScFilter*
-#   ecalLaserCorrFilter *
-#   trkPOGFilters
-   chargedHadronTrackResolutionFilter *
-   BadChargedCandidateFilter*
-   BadPFMuonFilter *
-   BadPFMuonDzFilter *
-   hfNoisyHitsFilter *
```
Comment (perrotta) [inline RecoMET/METFilters/python/metFilters_cff.py:122]: This is still said to be "work in progress": don't you plan to compute this bit as well, and then select on it at the analysis level only depending on the updated reccommandations?

**Comment by perrotta** [comment]
@lathomas could you please shortly summarize here what is the status of this PR, and plans and possible timeline for the updates?

**Comment by qliphy** [comment]
ping @lathomas

**Comment by mariadalfonso** [comment]
one observation:
this is the config files obtained from the `runTheMatrix -l 11634.0` in preparation of Run3
http://dalfonso.web.cern.ch/dalfonso/XPOG/Run3/step3_RAW2DIGI_L1Reco_RECO_RECOSIM_EI_PAT_VALIDATION_DQM.py
The MET Flag appear explicitly as `Path` in the final `Schedule` and not as part of the RECO nor PAT

I do not think it's normal, and maybe a good occasion to clean this as well

**Comment by perrotta** [comment]
ping @lathomas : please tell us what do you intend to do with this PR

**Comment by lathomas** [comment]
Hello. Sorry for the long silence. There is some recent update on the MET filter that I want to make sure I get right now. 
We will also need to play with modifiers since the recommendations differ across years and MC. Can you actually point to an example using a MC modifier (if such a thing exists)? 
Thanks

**Comment by slava77** [comment]
> Hello. Sorry for the long silence. There is some recent update on the MET filter that I want to make sure I get right now.
> We will also need to play with modifiers since the recommendations differ across years and MC. Can you actually point to an example using a MC modifier (if such a thing exists)?
> Thanks

https://github.com/cms-sw/cmssw/blob/master/RecoJets/JetProducers/python/PileupJetID_cfi.py#L39-L44
may give some idea.

**Comment by lathomas** [comment]
Thanks. This doesn't differentiate between data and MC however. Do you have an example for that? Or do I misread these lines?

**Comment by slava77** [comment]
> Thanks. This doesn't differentiate between data and MC however. Do you have an example for that? Or do I misread these lines?

would `miniAOD_customizeData`, `miniAOD_customizeMC`, and  `miniAOD_customizeAllMCFastSim` work for you?

**Comment by lathomas** [comment]
Ah yes there's an existing example for customizing the filters for FastSim actually so that should work ! Thanks.

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
@lathomas
- Are these recommendations now final? The [twiki](https://twiki.cern.ch/twiki/bin/viewauth/CMS/MissingETOptionalFiltersRun2) still reports about several "work in progress"...
- Should we expect some further update to this PR with the customization for (fast)sim that you anticipated in https://github.com/cms-sw/cmssw/pull/33818#issuecomment-873107491?

**Comment by lathomas** [comment]
Yes this is the current recommendation. It turns out the MC and data recommendation are now identical so no dedicated modifier is needed .

**Comment by perrotta** [comment]
+reconstruction
- List of MET filters updated with the latest recommendations from the JET/MET group
  - List is different for 2016 and 2017/18 data and MC: finally no specialization for MC or FastSim is foreseen
- Jenkins tests pass and only show differences in the events passing one single path of the TriggerResults, the one triggered by the bool resulting from the union of those recommended filters

**Pull request merged**